src/ucc/api/ucc.h

```diff
@@ -1333,25 +1333,38 @@ ucc_status_t ucc_team_get_all_eps(ucc_team_h team, uint64_t **ep,
  *  @ingroup UCC_COLLECTIVES_DT
  */
 typedef enum {
-    UCC_COLL_BUFF_FLAG_IN_PLACE             = UCC_BIT(0),
-    UCC_COLL_BUFF_FLAG_PERSISTENT           = UCC_BIT(1),
-    UCC_COLL_BUFF_FLAG_COUNT_64BIT          = UCC_BIT(2),
-    UCC_COLL_BUFF_FLAG_DISPLACEMENTS_64BIT  = UCC_BIT(3)
-} ucc_coll_buffer_flags_t;
+    UCC_COLL_ARGS_FLAG_IN_PLACE             = UCC_BIT(0),
+    UCC_COLL_ARGS_FLAG_PERSISTENT           = UCC_BIT(1),
+    UCC_COLL_ARGS_FLAG_COUNT_64BIT          = UCC_BIT(2),
+    UCC_COLL_ARGS_FLAG_DISPLACEMENTS_64BIT  = UCC_BIT(3)
+} ucc_coll_args_flags_t;
 
 /**
  *  @ingroup UCC_COLLECTIVES_DT
  */
+typedef enum ucc_memory_type {
+    UCC_MEMORY_TYPE_HOST,         /**< Default system memory */
+    UCC_MEMORY_TYPE_CUDA,         /**< NVIDIA CUDA memory */
+    UCC_MEMORY_TYPE_CUDA_MANAGED, /**< NVIDIA CUDA managed memory */
+    UCC_MEMORY_TYPE_ROCM,         /**< AMD ROCM memory */
+    UCC_MEMORY_TYPE_ROCM_MANAGED, /**< AMD ROCM managed system memory */
```

Comment:
should we change the comments text format  to "` /*!<` " to match with other places in this header file

Reply:
Yes. This is required for generating description for the fields in the doxygen.

Reply:
done